R/import_ebird.R

```diff
@@ -13,47 +13,75 @@
 #' tar <- ebird_sample_data()
 #' import_ebird(tar)
 #' 
-import_ebird <- function(tarfile){
+import_ebird <- function(tarfile) {
   
   source_dir <- tempfile("ebird_tmp")
   dest <- file.path(ebird_data_dir(), "parquet")
   
   dir.create(source_dir, recursive = TRUE)
   utils::untar(tarfile = tarfile, exdir = source_dir)
-  ebd <-list.files(source_dir, pattern="ebd.*\\.txt\\.gz", 
+  ebd <- list.files(source_dir, pattern = "ebd.*\\.txt\\.gz",
                    full.names = TRUE, recursive = TRUE)
   
-  
   ds <- arrow_open_ebird_txt(ebd, dest)
   
+  # clean up column names
+  col_names <- names(ds)
+  names(col_names) <- gsub("[/ ]", "_", tolower(col_names))
+  ds <- dplyr::select(ds, dplyr::all_of(col_names))
+  
   # Consider alternative partitions that might speed common queries
   # partitioning = c("COUNTRY")
   # However, arrow supports at most 1024 partitions
   
-  arrow::write_dataset(ds, dest, format="parquet")
+  arrow::write_dataset(ds, dest, format = "parquet")
   
+  unlink(source_dir, recursive = TRUE)
```

Comment:
Meant to put this in a separate PR, but just realized deleting the temp txt.gz file as per Issue #3 ended up in this PR. Do you want me to separate it out or OK to just leave it here.